ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/events/EventsRange.scala

```diff
@@ -61,4 +66,21 @@ object EventsRange {
       .as(get[Long](1).singleOpt)(connection)
       .getOrElse(EmptyLedgerEventSeqId)
   }
+
+  private[events] def readPage[A](
+      fasterRead: Long => SimpleSql[Row], // takes guessedPageEnd
+      saferRead: Int => SimpleSql[Row], // takes minPageSize
+      row: RowParser[A],
+      range: EventsRange[Long],
+      pageSize: Int): SqlSequence[Vector[A]] = {
+    val minPageSize = 10 min pageSize max (pageSize / 10)
+    val guessedPageEnd = range.endInclusive min (range.startExclusive + pageSize)
+    SqlSequence.vector(fasterRead(guessedPageEnd) withFetchSize Some(pageSize), row) flatMap {
+      arithPage =>
+        if (guessedPageEnd == range.endInclusive || arithPage.sizeIs >= minPageSize)
+          SqlSequence point arithPage
```

Comment:
I don't understand what's the principle that you use to decide whether or not to use infix notation. What is it? In general, every other call in this Bazel package uses infix notation for symbolic operators, otherwise the usual dot notation is used.

Reply:
The principle I tend to use is "do the . and () actually disambiguate anything, permit sensible code formatting (e.g. newlines in chains of calls), or do they exist solely to make it look more like Java?" If the latter, I use infix when it looks nicer.

This strategy can fairly be accused of "radical novelty", but I suggest this merely mirrors the "reactionary-ism" of the other strategy you imply. Consider `tx isReplayedBy replay(tx)`. Is this expression really made clearer by the form `tx.isReplayedBy(replay(tx))`? Or is that merely a hypothetical reader betrayed by their own expectation of seeing a language that is not Scala? This is what I regularly ask myself.

Since I'm staring at this code carefully, I would probably put a `.` before `flatMap`, which passes (2) (permit a sensible reformatting), but the others fall into (3).

Reply:
That's fine, I don't care too much about consistency at this level or using a specific convention, I was just unable to recognize the pattern.